src/include/platform/ConfigurationManager.h

```diff
@@ -89,6 +89,7 @@ class ConfigurationManager
     };
 
     virtual CHIP_ERROR GetPrimaryMACAddress(MutableByteSpan buf)                           = 0;
+    virtual CHIP_ERROR GetPrimaryEthernetMACAddress(MutableByteSpan buf)                   = 0;
```

Comment:
this adds a new method to ALL platforms ... what if platforms do not have an ethernet?

In particular this makes the fake platform fail compilation.

Reply:
but if we have an API for Wifi and Thread we should have one for ethernet, right?
It can be empty implementation if the platforms do not support ethernet

Reply:
The empty implementation is added to the Generic Configuration Manager (similar to Wifi and thread) and fake platform
Now the CI will also pass

Reply:
Resolving this thread as we decided to add an separate API for ethernet